marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/ExecEndpointImpl.java

```diff
@@ -166,32 +171,78 @@ private boolean processOutput(CallContextImpl callContext){
                         callContext.withEndpointState(output);
                     }
                     incrementCallCount();
-                } catch(Throwable throwable) {
+                    return output;
+                } catch(Throwable catchedThrowable) {
                     // TODO: logging
-                    throw new RuntimeException("error while calling "+getEndpoint().getEndpointPath(), throwable);
+                    throwable = catchedThrowable;
                 }
-// TODO -- retry with new session if times out
+                // TODO -- retry with new session if times out
 
-                switch(getPhase()) {
-                    case INTERRUPTING:
-                        setPhase(WorkPhase.INTERRUPTED);
-                        logger.info("exec interrupted endpoint={} count={} work={}",
-                                callContext.getEndpoint().getEndpointPath(), getCallCount(), callContext.getWorkUnit());
-                        return false;
-                    case RUNNING:
-                        if (output == null ) {
-                            if(getCallerThreadPoolExecutor() == null || getCallerThreadPoolExecutor().getActiveCount() <= 1)
-                                setPhase(WorkPhase.COMPLETED);
-                            logger.info("exec completed endpoint={} count={} work={}",
-                                    callContext.getEndpoint().getEndpointPath(), getCallCount(), callContext.getWorkUnit());
-                            return false;
+                if (throwable != null) {
+                    if (getErrorListener() == null) {
+                        logger.error("Error while calling " + getEndpoint().getEndpointPath(), throwable);
+                        throw new RuntimeException("Error while calling " + getEndpoint().getEndpointPath(), throwable);
```

Comment:
I'm wondering if it might be better to stop all calls if there is no error listener.  If we throw a only runtime exception in a thread, it will get thrown away. Also, that would be more consistent with the case where the error listener throws an error.

What do you think?

At the risk of pointing out the obvious. if we make the change, we'd want to make it in all of the callers.

Reply:
Yes, that's more consistent. Now the IF clause is changed as below.

if (getErrorListener() == null) {
                        logger.error("No error listener set. Stop all calls. " + getEndpoint().getEndpointPath(), throwable);
                        error = ErrorDisposition.STOP_ALL_CALLS;
                    }